File: src/template/template.go
package main

import (
	"bufio"
	"flag"
	"fmt"
	"log"
	"os"
)

var (
	partA       = flag.Bool("partA", true, "Perform part A solution?")
	inputFile   = flag.String("inputFile", "day!DAY!/input.txt", "Input File")
)

func main() {
	flag.Parse()

	inputText := processInput()
	
	if *partA {
		// part A
		fmt.Printf("Part A")
	} else {
		// part B
	}
}

func processInput() string {
    file, err := os.Open(*inputFile)
	if err != nil {
		log.Fatal(err)
	}
	//noinspection GoUnhandledErrorResult
	defer file.Close()

	// Read in inputs
	scanner := bufio.NewScanner(file)
	inputStringFromFile := ""
	for scanner.Scan() {
		inputStringFromFile += scanner.Text()
	}
    return inputStringFromFile
}

File: src/template/template.ts
import { inputAsStringArray } from '../utils/InputProcessor';

export const part1 = (input: string[]): number => {
  return input.length;
};

export const part2 = (input: string[]): number => {
  return input.length;
};

const input = inputAsStringArray('src/day!DAY!/input.txt');
console.log(
  `Day !DAY!:\n  Part 1:  ${part1(input)}\n  Part 2:  ${part2(input)}`
);
